go-selinux/pwalk.go

```diff
@@ -0,0 +1,128 @@
+// +build selinux,linux
+
+package selinux
+
+/* The code below was taken from
+ * https://github.com/stretchr/powerwalk/blob/master/walker.go
+ * and modified to use github.com/karrick/godirwalk
+ */
+
+import (
+	"sync"
+
+	"github.com/karrick/godirwalk"
+	"github.com/pkg/errors"
+)
+
+// DefaultConcurrentWalks is the default number of files that will be walked at the
+// same time when the Walk function is called.
+// To use a value other than this one, use the WalkLimit function.
+const DefaultConcurrentWalks int = 100
```

Comment:
A tool can make its own decision by using `WalkLimit()` function, which has this as a parameter. This is just a default, which I questioned as well but found sane enough.

I played a bit with various values for concurrency (on my laptop -- 8x cores, SSD), and they don't affect the performance much as long as the value is not less than the number of CPUs.
So, `100` looks like a sane default.

I'm not sure if it would be drastically different for rotating disks, as _I guess_ most if not all of the info required is already in the kernel's dentry cache. This would be interesting to play with though.

It might make sense to make this equal to e.g. `runtime.NumCPU() * 2`, iow

```go
func Walk(root string, options *godirwalk.Options) error {
￼	return WalkLimit(root, options, runtime.NumCPU() * 2)
￼}
```